src/commands/projects-create.ts

```diff
@@ -0,0 +1,59 @@
+import * as clc from "cli-color";
+import * as Command from "../command";
+import * as FirebaseError from "../error";
+import { createFirebaseProject, ParentResourceType } from "../firebase-resource-manager";
+import { prompt } from "../prompt";
+import * as requireAuth from "../requireAuth";
+import logger = require("../logger");
+
+module.exports = new Command("projects:create [projectId]")
+  .description("create a new firebase project")
+  .option("-n, --display-name <displayName>", "(optional) display name for the project")
+  .option(
+    "-o, --organization <organizationId>",
+    "(optional) ID of the parent Google Cloud Platform organization under which to create this project"
+  )
+  .option(
+    "-f, --folder <folderId>",
+    "(optional) ID of the parent Google Cloud Platform folder in which to create this project"
+  )
+  .before(requireAuth)
+  .action(async (projectId: string, options: any) => {
```

Comment:
Shouldn't this be `projectId: string | undefined` or `string | null` since it's an optional argument? Not sure which is correct, since I'm not sure which actually gets piped through when it's omitted.

Reply:
Changed to `projectId: string | undefined`